src/lib/index.js

```diff
@@ -350,19 +350,25 @@ lib.noneOrAll = function(containerIn, containerOut, attrList) {
     }
 };
 
-/** merge data array into calcdata items
+lib.mergeArray = function(traceAttr, cd, cdAttr) {
+    if(Array.isArray(traceAttr)) {
+        var imax = Math.min(traceAttr.length, cd.length);
+        for(var i = 0; i < imax; i++) cd[i][cdAttr] = traceAttr[i];
+    }
+};
+
+/** fill data array into calcdata items
  *
  * @param {array} traceAttr : trace attribute
  * @param {object} cd : calcdata trace
  * @param {string} cdAttr : calcdata key
  * @param {function} [fn] : optional function to apply to each array item
  */
-lib.mergeArray = function(traceAttr, cd, cdAttr, fn) {
+lib.fillArray = function(traceAttr, cd, cdAttr, fn) {
     fn = fn || lib.identity;
```

Comment:
🐎 I wonder whether it would be worth avoiding `fn` entirely if it's not provided? Either
`cd[i][cdAttr] = fn ? fn(traceAttr[i]) : traceAttr[i];`
or even
```
if(fn) {
    for(i = 0; i < cd.length; i++) cd[i][cdAttr] = fn(traceAttr[i]);
}
else {
    for(i = 0; i < cd.length; i++) cd[i][cdAttr] = traceAttr[i];
}
```